widgetastic_manageiq.py

```diff
@@ -425,3 +590,54 @@ class Accordion(PFAccordion):
     def is_dimmed(self):
         return bool(
             self.browser.elements('.//div[contains(@id, "tree") and contains(@class, "dimmed")]'))
+
+
+class Calendar(TextInput):
+    """A CFME calendar form field
+
+    Calendar fields are readonly, and managed by the dxhtmlCalendar widget. A Calendar field
+    will accept any object that can be coerced into a string, but the value may not match the format
+    expected by dhtmlxCalendar or CFME. For best results, either a ``datetime.date`` or
+    ``datetime.datetime`` object should be used to create a valid date field.
+
+    Args:
+        name: "name" property of the readonly calendar field.
+
+    Usage:
+
+        calendar = widgetastic_mangeiq.Calendar("miq_date_1")
+        calendar.fill(date(2000, 1, 1))
+        calendar.fill('1/1/2001')
+
+    """
+
+    def fill(self, value):
+        if isinstance(value, date):
+            date_str = '{}/{}/{}'.format(value.month, value.day, value.year)
+        else:
+            date_str = str(value)
+        self.browser.move_to_element
+        # need to write to a readonly field: resort to evil
+        if self.browser.get_attribute("ng-model", self) is not None:
+            # self.set_angularjs_value(self, date_str)
+            raise NotImplementedError
+        else:
+            self.browser.set_attribute(self, date_str)
+            # Now when we set the value, we need to simulate a change event.
+            if self.browser.get_attribute("data-date-autoclose", self):
+                # New one
+                script = "$(\"#{}\").trigger('changeDate');"
+            else:
+                # Old one
+                script = (
+                    "if(typeof $j == 'undefined') {var jq = $;} else {var jq = $j;} "
+                    "jq(\"#{}\").change();")
+            try:
+                self.browser.execute_script(script.format(self.name))
+            except WebDriverException as e:
+                logger.warning(
+                    "An exception was raised during handling of the Cal #{}'s change event:\n{}"
+                    .format(self.name, str(e)))
+        self.browser.plugin.ensure_page_safe()
+
+        return True
```

Comment:
Do we assume that the date is always changed, therefore we always return True? (you might want to check it first)